pkg/controller/containerjfr/containerjfr_controller_test.go

```diff
@@ -0,0 +1,563 @@
+
+	AfterEach(func() {
+		objs = nil
+	})
+
+	Describe("reconciling a request", func() {
+		Context("when reconciling", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewContainerJFR(),
+				}
+			})
+			It("should create certificates", func() {
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "containerjfr", Namespace: "default"}}
+				result, err := controller.Reconcile(req)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(result).To(Equal(reconcile.Result{RequeueAfter: 5 * time.Second}))
+				certNames := []string{"containerjfr", "containerjfr-ca", "containerjfr-grafana"}
+				for _, certName := range certNames {
+					cert := &certv1.Certificate{}
+					err := client.Get(context.Background(), types.NamespacedName{Name: certName, Namespace: "default"}, cert)
+					Expect(err).ToNot(HaveOccurred())
+				}
+			})
+			It("should create routes", func() {
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "containerjfr", Namespace: "default"}}
+				result, err := controller.Reconcile(req)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(result).To(Equal(reconcile.Result{RequeueAfter: 5 * time.Second}))
+
+				// Update certificate status
+				MakeCertificatesReady(client)
+				InitializeSecrets(client)
+
+				// Check for routes, ingress configuration needs to be added as each
+				// one is created so that they all reconcile successfully
+				result, err = controller.Reconcile(req)
+				IngressConfig(client, *controller, req, false)
+			})
+		})
+		Context("succesfully creates required resources", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewContainerJFR(),
+				}
+			})
+			It("should create persistent volume claim and set owner", func() {
+				pvc := &corev1.PersistentVolumeClaim{}
+				err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, pvc)
+				Expect(err).To(HaveOccurred())
+
+				ReconcileFully(client, *controller, false)
+
+				err = client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, pvc)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Compare to desired spec
+				expectedPvc := resource_definitions.NewPersistentVolumeClaimForCR(test.NewContainerJFR())
+				Expect(pvc.ObjectMeta.Name).To(Equal(expectedPvc.ObjectMeta.Name))
+				Expect(pvc.ObjectMeta.Namespace).To(Equal(expectedPvc.ObjectMeta.Namespace))
+				Expect(pvc.Spec.AccessModes).To(Equal(expectedPvc.Spec.AccessModes))
+				Expect(pvc.Spec.StorageClassName).To(Equal(expectedPvc.Spec.StorageClassName))
+
+				pvcStorage := pvc.Spec.Resources.Requests["storage"]
+				expectedPvcStorage := expectedPvc.Spec.Resources.Requests["storage"]
+				Expect(pvcStorage.Equal(expectedPvcStorage)).To(BeTrue())
+
+				// Check for owner
+				Expect(pvc.ObjectMeta.OwnerReferences[0].Kind).To(Equal("ContainerJFR"))
+				Expect(pvc.ObjectMeta.OwnerReferences[0].Name).To(Equal("containerjfr"))
```

Comment:
You could refactor this into a helper, along with other ObjectMeta checks. You should be able to use `metav1.Object` as the parameter type.